lib/ansible/modules/web_infrastructure/ansible_tower/tower_settings.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# (c) 2018, Nikhil Jain <nikjain@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: tower_settings
+author: "Nikhil Jain (@jainnikhil30)"
+version_added: "2.7"
+short_description: Get, Modify Ansible Tower settings.
+description:
+    - Get, Modify Ansible Tower settings. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name of setting to get/modify.
+    value:
+      description:
+        - Value to be modified for given setting. If this parameter is not given, module will only get the value of setting.
```

Comment:
This feels like it's somewhat filling the role of a facts module. Should these be separate?

Reply:
@wwitzel3 Thoughts on this?

Reply:
Hmm... I saw Alan's note that the API supports changing a setting to it's default value below.  If that's the case you might want to think about whether that's a useful thing to expose to the user here as well (it feels like it is but I'm not entirely sure).  If so, you should consider how it fits into these parameters.  None might be the best value to mean reset to default.  In which case the way to specify "show me the current value" would need to be something different.

Reply:
Hmm... I saw Alan's note that the API supports changing a setting to it's default value below.  If that's the case you might want to think about whether that's a useful thing to expose to the user here as well (it feels like it is but I'm not entirely sure).  If so, you should consider how it fits into these parameters.  None might be the best value to mean reset to default.  In which case the way to specify "show me the current value" would need to be something different.

Reply:
@abadger 
Thanks for your review. 
I think every settings field on Tower has a associated "default" value to it. The way "revert to default" works is, we have a "Revert" option on UI. Clicking that sends a PATCH with default value for that field in it. But I dont think tower-cli supports this yet, which is what we use in the tower modules. Once we have that in tower-cli, I think we can expose it to the modules too ?

Reply:
<nod>. Yeah, so if you're going to expose that in this module (which makes sense) then you need to decide on the UI for doing that now.  You don't have to implement it but you have to decide what it will be so that you don't do something now that prevents you from using it for what you want later.  For instance, if you want to use value=None for seeing the default in the future, you should make it so that specifying value=None right now is an error.

If you make value=None mean, "don't make any changes, just return the current value of the seeing" now then you'll need a different UI for seeing to default.